sgkit/tests/io/vcf/test_vcf_reader.py

```diff
@@ -1036,3 +1037,356 @@ def test_vcf_to_zarr__fields_errors(shared_datadir, tmp_path):
             fields=["INFO/AN"],
             field_defs={"INFO/AN": {"Type": "Blah"}},
         )
+
+
+@pytest.mark.parametrize(
+    "vcf_file, expected_sizes",
+    [
+        (
+            "sample.vcf.gz",
+            {
+                "max_alt_alleles": 3,
+                "field_defs": {"INFO/AC": {"Number": 2}, "INFO/AF": {"Number": 2}},
+                "ploidy": 2,
+            },
+        ),
+        ("mixed.vcf.gz", {"max_alt_alleles": 1, "ploidy": 4}),
+        ("no_genotypes.vcf", {"max_alt_alleles": 1}),
+        (
+            "CEUTrio.20.21.gatk3.4.g.vcf.bgz",
+            {
+                "max_alt_alleles": 7,
+                "field_defs": {"FORMAT/AD": {"Number": 8}},
+                "ploidy": 2,
+            },
+        ),
+    ],
+)
+def test_zarr_array_sizes(shared_datadir, vcf_file, expected_sizes):
+    path = path_for_test(shared_datadir, vcf_file)
+    sizes = zarr_array_sizes(path)
+    assert sizes == expected_sizes
+
+
+def check_field(group, name, ndim, shape, dimension_names, dtype):
+    assert group[name].ndim == ndim
+    assert group[name].shape == shape
+    assert group[name].attrs["_ARRAY_DIMENSIONS"] == dimension_names
+    if dtype == str:
+        assert group[name].dtype == np.object_
+        assert VLenUTF8() in group[name].filters
+    else:
+        assert group[name].dtype == dtype
+
```

Comment:
Doesn't look like there's any tests for the multiple FILTER behaviour?

Reply:
Done